content/tutorial-plotting-fractals.md

```diff
@@ -0,0 +1,558 @@
+
+Of course, we can do much better and design a function that performs the procedure on an entire mesh.
+
+```{code-cell} ipython3
+def divergence_rate(mesh,num_iter=10,radius=2): 
+        
+    z = mesh.copy()
+    diverge_len = np.zeros(mesh.shape) #Keep tally of the number of iterations
+        
+    for i in range(num_iter):
+        #Iterate on element if and only if |element| < radius (Otherwise assume divergence)
+        diverge_len[np.abs(z) < radius] += 1
+        z[np.abs(z) < radius] = f(z[np.abs(z)< radius])   
+        
+        
+    return diverge_len
+```
+
+The behaviour of this function may look confusing at first glance, so it will help to explain some of the notation.
+
+Our goal is to iterate over each value in the mesh and to tally the number of iterations before the value diverges. Since some values will diverge quicker than others, we need a procedure that only iterates over values that have an absolute value that is sufficiently small enough. We also want to stop tallying values once they surpass the radius. For this, we can use **[Boolean Indexing](https://numpy.org/devdocs/reference/arrays.indexing.html#boolean-array-indexing)**, a NumPy procedure that when paired with Universal Functions is unbeatable. Boolean Indexing allows for operations to be performed conditionally on a NumPy array without having to resort to looping over and checking for each array value individually. 
+
+In our case, we use a loop to apply iterations to our function $f(z) = z^2 -1 $ and keep tally. Using Boolean indexing, we only apply the iterations to values that have an absolute value less then 2. 
+
+With that out of the way, we can go about plotting our first fractal! We will use the [**imshow**](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.imshow.html) function to create a colour-coded visualization of the tallies. 
+
+```{code-cell} ipython3
+x,y = np.meshgrid(np.arange(-2,2,0.01),np.arange(-2,2,0.01))
+mesh =  x + (1j *y) 
+
+output = divergence_rate(mesh)
+
+fig = plt.figure(figsize=(5,5))
+ax = plt.axes()
+
+ax.set_title('$f(z) = z^2 -1$')
+ax.set_xlabel('Real axis')
+ax.set_ylabel('Imaginary axis')
+
+im = ax.imshow(output,extent=[-2,2,-2,2])
+divider = make_axes_locatable(ax)
+cax = divider.append_axes("right", size="5%", pad=0.1)
+plt.colorbar(im, cax=cax, label = 'Number of iterations');
+```
+
+What this stunning visual conveys is the complexity of the function’s behaviour. The yellow region represents values that remain small, while the purple region represents the divergent values. The beautiful pattern that arises on the border of the converging and diverging values is even more fascinating when you realize that it is created from such a simple function.
+
++++
+
+# Mandelbrot 
+
+What we just explored was an example of a fractal found in the Mandelbrot set. We can write a more generalized version of the above function to gain access to a wider range of fractals.
```

Comment:
Really glad you brought this up, I learned a lot of what I know about fractals by writing this tutorial, somehow missed this connection! Before my next commit, I will do some additional reading on Julia sets and will modify the tutorial to briefly explain what it is. Luckily this should be relatively easy to update since there are only two sections with the "general Mandelbrot function" where the first section actually is plotting the Julia set. Thanks for bringing this up, it would have been a shame to have had this slip past revision!